src/source/src/connector_source.rs

```diff
@@ -99,14 +97,21 @@ impl SourceChunkBuilder for ConnectorStreamReader {}
 
 #[async_trait]
 impl StreamSourceReader for ConnectorStreamReader {
-    async fn next(&mut self) -> Result<StreamChunk> {
+    async fn next(&mut self) -> Result<StreamChunkWithState> {
         match self.reader.next().await.to_rw_result()? {
-            None => Ok(StreamChunk::default()),
+            None => Ok(StreamChunkWithState {
+                chunk: StreamChunk::default(),
+                split_offset_mapping: HashMap::new(),
+            }),
             Some(batch) => {
                 let mut events = Vec::with_capacity(batch.len());
+                let mut split_offset_mapping: HashMap<String, String> = HashMap::new();
 
                 for msg in batch {
                     if let Some(content) = msg.payload {
+                        *split_offset_mapping
+                            .entry(msg.split_id.clone())
+                            .or_insert("".to_string()) = msg.offset.to_string();
```

Comment:
How about...
```suggestion
                        split_offset_mapping.insert(msg.split_id, msg.offset.to_string());
```

Reply:
in most cases, the hashmap will be updated rather than inserted.

Reply:
I think the two ways are exactly the same, except for allocating an unused empty string.